managegroup.php

```diff
@@ -257,185 +263,141 @@
-		}
-
-		eval("\$users .= \"".$templates->get("managegroup_user")."\";");
-	}
-
-	if(!$users)
-	{
-		eval("\$users = \"".$templates->get("managegroup_no_users")."\";");
-	}
-
-	$add_user = '';
-	$remove_users = '';
-	if($groupleader['canmanagemembers'] == 1)
-	{
-		eval("\$add_user = \"".$templates->get("managegroup_adduser")."\";");
-		eval("\$remove_users = \"".$templates->get("managegroup_removeusers")."\";");
-	}
-
-	if($usergroup['type'] == 5 && $groupleader['caninvitemembers'] == 1)
-	{
-		eval("\$invite_user = \"".$templates->get("managegroup_inviteuser")."\";");
-	}
-
-	$plugins->run_hooks("managegroup_end");
-
-	eval("\$manageusers = \"".$templates->get("managegroup")."\";");
-	output_page($manageusers);
+    $plugins->run_hooks("managegroup_start");
+
+    $lang->members_of = $lang->sprintf($lang->members_of, htmlspecialchars_uni($usergroup['title']));
+    $lang->add_member = $lang->sprintf($lang->add_member, htmlspecialchars_uni($usergroup['title']));
+    $lang->invite_member = $lang->sprintf($lang->invite_member, htmlspecialchars_uni($usergroup['title']));
+    $usergroup['pendingrequests'] = false;
+
+    if ($usergroup['type'] == 5) {
+        $usergroup['usergrouptype'] = $lang->group_public_invite;
+    }
+    elseif ($usergroup['type'] == 4) {
+        $query = $db->simple_select("joinrequests", "COUNT(*) AS req", "gid='{$gid}'");
+        $numrequests = $db->fetch_array($query);
+
+        if ($numrequests['req']) {
+            $usergroup['pendingrequests'] = true;
+            $lang->num_requests_pending = $lang->sprintf($lang->num_requests_pending, $numrequests['req']);
+        }
+        $usergroup['usergrouptype'] = $lang->group_public_moderated;
+    }
+    elseif ($usergroup['type'] == 3) {
+        $usergroup['usergrouptype'] = $lang->group_public_not_moderated;
+    }
+    elseif ($usergroup['type'] == 2) {
+        $usergroup['usergrouptype'] = $lang->group_private;
+    } else {
+        $usergroup['usergrouptype'] = $lang->group_default;
+    }
+
+    $usergroup['leaders'] = false;
+
+    // Display group leaders (if there is any)
+    $query = $db->query("
+        SELECT g.*, u.username, u.usergroup, u.displaygroup
+        FROM " . TABLE_PREFIX . "groupleaders g
+        LEFT JOIN " . TABLE_PREFIX . "users u ON (u.uid=g.uid)
+        WHERE g.gid = '{$gid}'
+    ");
+
+    $leaders_array = array();
+
+    if ($db->num_rows($query)) {
+        $loop = 1;
+        $leaders = [];
+        $leader_count = $db->num_rows($query);
+        while ($leader = $db->fetch_array($query)) {
+            $leader['username'] = htmlspecialchars_uni($leader['username']);
+            $leader_name = format_name($leader['username'], $leader['usergroup'], $leader['displaygroup']);
+            $leader['profilelink'] = build_profile_link($leader_name, $leader['uid']);
+
+            $leaders_array[] = $leader['uid'];
+
+            // Get commas...
+            if ($loop != $leader_count) {
```

Comment:
This logic could be done in the view itself, using [the `loop.last` variable](https://twig.symfony.com/doc/2.x/tags/for.html#the-loop-variable).